src/analyses/base.ml

```diff
@@ -404,20 +404,12 @@ struct
     | _, Bot -> Bot
     | _ -> VD.top ()
 
-  (* Auxiliary function to append an additional offset to a given offset. *)
-  let rec add_offset ofs add =
-    match ofs with
-    | `NoOffset -> add
-    | `Field (fld, `NoOffset) -> `Field (fld, add)
-    | `Field (fld, ofs) -> `Field (fld, add_offset ofs add)
-    | `Index (exp, `NoOffset) -> `Index (exp, add)
-    | `Index (exp, ofs) -> `Index (exp, add_offset ofs add)
-
+  (* TODO: move to Lval *)
```

Comment:
Where exactly is this supposed to be moved to?

Reply:
I've added longer explanations to the TODOs you've mentioned here. Mostly the idea is to also use `AddressDomain` for queries, such that all those lossy and hacky back-and-forth conversions could be avoided. But I don't want to make this PR bigger than it already is.